File: tests/fixtures/objc/excess_args.m
#import <Foundation/Foundation.h>

@implementation UserService

- (BOOL)createUser:(NSString *)name
             email:(NSString *)email
               age:(NSInteger)age
              role:(NSString *)role
        department:(NSString *)department
            salary:(double)salary {
    NSLog(@"Creating user %@", name);
    return YES;
}

- (instancetype)initWithName:(NSString *)name
                       email:(NSString *)email
                         age:(NSInteger)age
                        role:(NSString *)role
                  department:(NSString *)department
                      salary:(double)salary
                      extra7:(id)extra7
                      extra8:(id)extra8
                      extra9:(id)extra9 {
    self = [super init];
    if (self) {
        _name = name;
    }
    return self;
}

- (NSString *)getName {
    return _name;
}

@end

File: tests/fixtures/swift/excess_args.swift
func createUser(name: String, email: String, age: Int, role: String, department: String, salary: Double) -> Bool {
    return true
}

func simpleFunc(x: Int) -> Int {
    return x + 1
}

class Config {
    var host: String = ""
    var port: Int = 0
    var timeout: Int = 0
    var retries: Int = 0
    var verbose: Bool = false
    var secure: Bool = false

    init(host: String, port: Int, timeout: Int, retries: Int, verbose: Bool, secure: Bool, p7: Int, p8: Int, p9: Int) {
        self.host = host
        self.port = port
        self.timeout = timeout
        self.retries = retries
        self.verbose = verbose
        self.secure = secure
    }
}
